src/App.tsx

```diff
@@ -1,14 +1,53 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useContext, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodosContext } from './contexts/TodoProvider';
+import { FilterOptions } from './types/FilterOptions';
+import { FilterStatus } from './types/FilterStatus';
+import { Todo } from './types/Todo';
+import { TodoModal } from './components/TodoModal';
+
+function filterTodos(todos: Todo[], { query, status }: FilterOptions) {
+  let filteredTodos = [];
+
+  switch (status) {
+    case FilterStatus.Active:
+      filteredTodos = todos.filter(({ completed }) => !completed);
+      break;
+    case FilterStatus.Completed:
+      filteredTodos = todos.filter(({ completed }) => completed);
+      break;
+    default:
+      filteredTodos = [...todos];
+      break;
+  }
+
+  if (query) {
+    const normalizedQuery = query.trim().toLowerCase();
+
+    filteredTodos = filteredTodos
+      .filter(({ title }) => title.toLowerCase().includes(normalizedQuery));
+  }
+
+  return filteredTodos;
+}
```

Comment:
you can move it outside component file.
If you can write constant instead let use it prefer.
let worsens the readability and perception of the code during scaling